lib/model/game_state.dart

```diff
@@ -35,11 +44,25 @@ class GameState {
   }
 
   void scryCards(int numToScry) {
-    //TODO - implement scrying
+    final actualNumToScry = min(deck.length, numToScry);
+    scrying.addAll(deck.getRange(0, actualNumToScry));
+    deck.removeRange(0, actualNumToScry);
+  }
+
+  void replaceScryedCard(Card card, ScryOption position) {
+    scrying.remove(card);
+
+    if (position == ScryOption.TOP) {
+      deck.insert(0, card);
+    } else {
+      deck.add(card);
+    }
   }
 
   void exileCurrentCard() {
```

Comment:
Could test for the case where current card is null?

Reply:
I wish Dart had non nullable types in a proper way! `currentCard` should never be null except at the beginning - but we could even arrange so that's not the case by making `GameState` automatically shuffle and draw a card whenever it's constructed. 

So I'm hoping there's no case to test here, really. All I could do would be to throw an exception which... ugh. I wish the compiler could just do it for us :/